Feature/fix linux amd performance

soulsource opened the pull request:
This commit replaces the hardcoded Bayer pattern in dither() with a function that computes those numbers on the fly.
The performance on Linux with AMD hardware is vastly improved by this change, as it prevents usage of Private Memory, and keeps all data in the registers. However it needs to be tested on other platforms, as it might have a negative performance impact...

Note: This currently requires SM 4.

LGhassen commented:
1) Is there a reason this requires SM 4.0? It's important to keep compatibility with SM 3.0

2) Why is the PR closed?

soulsource commented:
The pull request is closed because this needs SM 4, and I'm currently working on a solution that does not.

LGhassen commented:
What if we use a lookup texture instead of the pattern?

soulsource commented:
Should probably work as well. Anyhow, I'm nearly done with removing the stuff that requires SM 4 features. Just testing it once more.

soulsource commented:
If a lookup texture or the code that calculates the points is faster would probably require testing.

soulsource commented:
So, I've removed all bitwise operators (SM 4...), and also switched everything over to floating point. That should make this work with SM 3. At least now the Windows build passes without any warnings or errors.

LGhassen commented:
I will test it, not sure I can reproduce the original slowdown though, which GPU do you have?

soulsource commented:
It's an AMD Rx 480. As said, I'm using the AMD open source drivers on Linux. The shader compiler that's currently installed on my system is llvm-6.0.1.

LGhassen commented:
Hi @soulsource, for future updates of Scatterer I'm thinking of splitting the mod into an open source part (the plugin code which will continue as it is now to be GPL) and a closed source part (the shaders which will be provided in compiled/assetbundle form only). Would you be ok if I use your contribution in the closed source shaders?

soulsource commented:
Hi!

 

Yes, of course. Feel free to use those few commits however you like. It’s anyhow just the recursive formula for a Bayer pattern taken from Wikipedia and rewritten in iterative form.

Thanks for asking!

 

Best Regards,

Andreas

 

Von: LGhassen ***@***.***> 
Gesendet: Dienstag, 15. Februar 2022 12:32
An: LGhassen/Scatterer ***@***.***>
Cc: Andreas Grois ***@***.***>; Mention ***@***.***>
Betreff: Re: [LGhassen/Scatterer] Feature/fix linux amd performance (#52)

 

Hi @soulsource <https://github.com/soulsource> , for future updates of Scatterer I'm thinking of splitting the mod into an open source part (the plugin code which will continue as it is now to be GPL) and a closed source part (the shaders which will be provided in compiled/assetbundle form only). Would you be ok if I use your contribution in the closed source shaders?

—
Reply to this email directly, view it on GitHub <https://github.com/LGhassen/Scatterer/pull/52#issuecomment-1040165464> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AA23BVCNPW5WSAUXGCIGEBLU3I2SFANCNFSM4GKTL6AA> .
Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>  or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub> . 
You are receiving this because you were mentioned.  <https://github.com/notifications/beacon/AA23BVEXMKXID6JW4HAMLJTU3I2SFA5CNFSM4GKTL6AKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOHX72UWA.gif> Message ID: ***@***.*** ***@***.***> >
